hal/armv7a/imx6ull/imx6ull.c

```diff
@@ -426,4 +427,8 @@ void _hal_platformInit(void)
 
 	/* kick watchdog power down counter */
 	*(imx6ull_common.wdog + wdog_wmcr) = 0;
+
+	/* copy watchdog Reset Status Register to bootreason[23:16] */
+	imx6ull_bootReason &= 0xff00ffffu;
+	imx6ull_bootReason |= *(imx6ull_common.wdog + wdog_wrsr) << 16;
```

Comment:
Are certain that other bits are zero? If so ignore my suggestion

```suggestion
	imx6ull_bootReason |= (*(imx6ull_common.wdog + wdog_wrsr) & 0xff) << 16;
```